### PR TITLE
Reports: Add traditional JSON report

### PR DESCRIPTION
**Pull request by psiinon**
Depends on #2951 - will refactor this once thats merged.

### CODE REVIEW

File: addOns/reports/CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,15 @@ All notable changes to this add-on will be documented in this file.
...
 
 ## Unreleased
+### Added
+- Traditional JSON report
+
 ### Changed
 - Maintenance changes.
 
+### Fixed
+- Include all relevant alerts in XML report templates (Issue 6627).
```
Comment (kingthorin) [inline addOns/reports/CHANGELOG.md:14]: Needs a rebase?

Comment (kingthorin) [reply addOns/reports/CHANGELOG.md:14]: Oh crap, missed it. Sorry.

**Comment by psiinon** [comment]
Now refactored and ready for review.

The JSON reports should be equivalent to the existing core ones with the
exception of the formating. The escaping should be exactly the same.
The aim is to ensure anyone who relies on these reports should not be
impacted when we switch over.

I'll need to make similar changes to the XML report.

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java [hunk 2]
```diff
@@ -124,4 +126,26 @@ public static boolean hasSiteStats(String site, String prefix) {
...
         return list;
     }
+
+    /** A method which mimics the escaping used for traditional ZAP reports */
+    public String replaceEntity(String text) {
+        String result = text;
+        if (text != null) {
+            result = StringEscapeUtils.escapeXml10(XMLStringUtil.escapeControlChrs(result));
+        }
+        return result;
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java:136]: Do away with the intermediate storage?
```java
            return text != null ? StringEscapeUtils.escapeXml10(XMLStringUtil.escapeControlChrs(text)) : text;
```

Comment (psiinon) [reply addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportHelper.java]: Actually both of those methods handle null so no need to even do the null test, so updated to remove that :)

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**